android/src/main/java/com/guichaguri/trackplayer/service/models/Track.java

```diff
@@ -166,20 +176,27 @@ public DataSource createDataSource() {
             // Creates a local source factory
             ds = new DefaultDataSourceFactory(ctx, userAgent);
 
+        } else if ( icymetadata ) {
+            OkHttpClient client = new OkHttpClient.Builder().build();
+            IcyHttpDataSourceFactory icyDataSource = new IcyHttpDataSourceFactory.Builder(client)
+                            .setUserAgent(userAgent)
+                            .setIcyHeadersListener(IcyStreamUpdate.HeaderUpdate)
+                            .setIcyMetadataChangeListener(IcyStreamUpdate.MetaDataUpdate)
+                            .build();
+            ds = new DefaultDataSourceFactory(ctx, null, icyDataSource);
+            ds = playback.enableCaching(ds);
         } else {
-
             // Creates a default http source factory, enabling cross protocol redirects
             ds = new DefaultHttpDataSourceFactory(
                     userAgent, null,
                     DefaultHttpDataSource.DEFAULT_CONNECT_TIMEOUT_MILLIS,
                     DefaultHttpDataSource.DEFAULT_READ_TIMEOUT_MILLIS,
                     true
             );
-
             ds = playback.enableCaching(ds);
-
```

Comment:
Don't remove those empty lines, they are meant to help reading the code.

Reply:
Sorry, custom to remove the end lines and keep in the beginning.